src/ucp/dt/datatype_iter.inl

```diff
@@ -189,6 +191,43 @@ ucp_datatype_iter_cleanup(ucp_datatype_iter_t *dt_iter, unsigned dt_mask)
     }
 }
 
+/*
+ * Check if the iterator has start position
+ */
+static UCS_F_ALWAYS_INLINE int
+ucp_datatype_iter_is_begin(const ucp_datatype_iter_t *dt_iter)
+{
+    return dt_iter->offset == 0;
+}
+
+/*
+ * Restart iterator @a dt_iter to begin state.
+ *
+ * @note The function invokes @ref ucp_generic_dt_ops_t::start_pack callback for
+ *       generic data type and assumes that all data types are in initial
+ *       position beause UCP currently does not support restart of request in
+ *       the middle state.
```

Comment:
The comment should describe assumptions and not implementation so it should be generic and not what happens for each data type:
```
Initialize a cleaned-up iterator, after it has already been initialized and cleaned up by 
calling @ref ucp_datatype_iter_init and @ref ucp_datatype_iter_cleanup.
@note The current offset of the iterator must be 0, since we can't start the iterator from the middle.
```